### PR TITLE
Response Server UI

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
Test changes supporting consolidating settings and removing dependency on Folder Management Module Properties feature, improving usability by offering instead a more clear form with added validation and guidance.

#### Related Pull Requests
* https://github.com/FDA-MyStudies/Response/pull/28
* https://github.com/LabKey/platform/pull/2492

#### Changes
* Add helper to reach Response Server Admin UI page
* Add helper for setting Response Server Configurations

### CODE REVIEW

File: src/org/labkey/test/BaseWebDriverTest.java [hunk 1]
```diff
@@ -1552,6 +1558,30 @@ public String getModulePropertyValue(ModulePropertyValue property)
         return (String)ref.getValue();
     }
 
+    public void setResponseServerConfigurations(LinkedHashMap<String, String> props)
```
Comment (labkey-adam) [inline src/org/labkey/test/BaseWebDriverTest.java]: I'll let @labkey-tchad weigh in on this, but it doesn't seem right that `BaseWebDriverTest` should know anything about the response server config details. It knows about module properties, but that's a very generic mechanism.

File: src/org/labkey/test/pages/core/admin/ShowAdminPage.java [hunk 2]
```diff
@@ -141,6 +141,12 @@ public LookAndFeelSettingsPage clickLookAndFeelSettings()
         return new LookAndFeelSettingsPage(getDriver());
     }
 
+    public void clickResponseServerConfiguration()
```
Comment (labkey-adam) [inline src/org/labkey/test/pages/core/admin/ShowAdminPage.java:144]: I guess there's precedent for `ShowAdminPage` knowing about all the module-specific links...

File: src/org/labkey/test/BaseWebDriverTest.java [hunk 3]
```diff
@@ -1552,6 +1558,30 @@ public String getModulePropertyValue(ModulePropertyValue property)
...
+            else
+                setFormElement(Locator.name(prop), val);
+        }
+        clickButton("Save and Finish", 0);
+
+        assertTextNotPresent("WCP Base URL must begin with 'http://' or 'https://'");
+        assertTextNotPresent("WCP Base URL must end with '/StudyMetaData'");
+        assertTextNotPresent("WCP username must not be blank");
+        assertTextNotPresent("WCP password must not be blank");
+        assertTextNotPresent("Metadata Directory path is invalid");
```
Comment (labkey-tchad) [inline src/org/labkey/test/BaseWebDriverTest.java]: Checking for the non-existence of text is not reliable (somebody might change the wording of an error).
You should be able to do something like `assertElementNotPresent(Locators.labkeyError)` to catch any of these.

File: src/org/labkey/test/BaseWebDriverTest.java [hunk 4]
```diff
@@ -1552,6 +1558,30 @@ public String getModulePropertyValue(ModulePropertyValue property)
...
+        {
+            String val = props.get(prop);
+            log("setting property: " + prop + " to value: " + val);
+
+            if (prop.equals("metadataLoadLocation"))
+                checkRadioButton(Locator.radioButtonByNameAndValue("metadataLoadLocation", val));
+            else
+                setFormElement(Locator.name(prop), val);
+        }
+        clickButton("Save and Finish", 0);
```
Comment (labkey-tchad) [inline src/org/labkey/test/BaseWebDriverTest.java]: Does "Save and Finish" not navigate? Failing to wait for page loads that occur is a frequent source of intermittent failures.
```suggestion
        clickButton("Save and Finish");
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
Adam is correct. Except for the most core modules; module-specific test functionality should live in that module. I left some feedback here but these changes should go in the Response repository in a helper class or a page class for `ServerConfigurationAction`

File: src/org/labkey/test/BaseWebDriverTest.java (hunk 3 above)
Comment (RosalineP) [reply src/org/labkey/test/BaseWebDriverTest.java]: Ahhh makes sense, done!

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
Adam is correct. Except for the most core modules; module-specific test functionality should live in that module. These changes should go in the Response repository in a helper class or a page class for ServerConfigurationAction.
This branch can be deleted.

**Comment by RosalineP** [comment]
Changes moved to `Response` branch!

**Pull request closed without merging**